File: frontend/src/models/Repair.ts
import { CustomerModel } from "./CustomerModel";
import { PaymentModel } from "./OrderModel";
type RepairStatus = "pending" | "completed" | "canceled";

interface Repair {
  customer: 1;
  repair_issue: string;
  status: string;
  note: string;
  cost: number;
  total_price: number;
  payment_amount: number;
}
interface RepairResponse {
  id: number;
  customer: CustomerModel;
  repair_issue: string;
  status: RepairStatus;
  note: string;
  cost: number;
  total_price: number;
  balance: number;
  payments: PaymentModel[];
  created_at: string;
}
export type { Repair, RepairResponse };

File: mypos/api/models/repair.py
from django.db import models
from .customer import Customer

class Repair(models.Model):
    STATUS_CHOICES = [
        ("pending", "Pending"),
        ("completed", "Completed"),
        ("canceled", "Canceled"),
    ]
    customer=models.ForeignKey(Customer, on_delete=models.PROTECT)
    repair_issue = models.CharField(max_length=100,)  # ✅ Fixed typo
    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default="pending")
    note = models.TextField(blank=True, null=True)  # ✅ Changed from choices to TextField
    cost = models.IntegerField(default=0)
    total_price = models.IntegerField(default=0)
    balance = models.IntegerField(default=0)
    created_at = models.DateTimeField(auto_now_add=True)
    def __str__(self):
        return self.repair_issue  # ✅ Fixed return value
